applications/MultilevelMonteCarloApplication/tests/test_xmcAlgorithm.py

```diff
@@ -280,8 +298,15 @@ def test_mlmc_Kratos(self):
             estimations = get_value_from_remote(algo.estimation())
             estimated_mean = 1.47
             self.assertAlmostEqual(estimations[0], estimated_mean, delta=1.0)
-            for level in algo.hierarchy():
-                self.assertEqual(level[1], 15)
+            if "asynchronous_adaptivefixednumberlevels" in parametersPath:
+                self.assertAlmostEqual(sum(estimations), 1.5256974986866585, delta=parameters["monoCriteriaInputDictionary"]["statisticalError"]["tolerance"][0])
+                # self.assertEqual(sum(algo.hierarchy()[i][-1] for i in range (0,len(algo.hierarchy()))), 194) # uncomment once issue #62 is solved
+            elif "adaptivefixednumberlevels" in parametersPath:
+                self.assertAlmostEqual(sum(estimations), 1.5292773966722353, delta=parameters["monoCriteriaInputDictionary"]["statisticalError"]["tolerance"][0])
+                # self.assertEqual(sum(algo.hierarchy()[i][-1] for i in range (0,len(algo.hierarchy()))), 122) # uncomment once issue #62 is solved
```

Comment:
Does this need to be commented still?

Reply:
Yes, there is an open issue in XMC (see [issue 62](https://gitlab.com/RiccardoRossi/exaqute-xmc/-/issues/62)) where I point out that adaptive estimation of levels and of samples per level is not fixed (when a given seed is provided). I prefer to keep this code commented here to know that once the issue is solver we can uncomment these lines.